url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParser.java

```diff
@@ -0,0 +1,3669 @@
+             modifier="all ";
+          break;
+        case K_DISTINCT:
+          jj_consume_token(K_DISTINCT);
+                                               modifier="distinct ";
+          break;
+        default:
+          jj_la1[93] = jj_gen;
+          jj_consume_token(-1);
+          throw new ParseException();
+        }
+        s2 = TableColumn();
+        jj_consume_token(90);
+       {if (true) return new MExpression(s, new MConstant(modifier + s2, MConstant.COLUMNNAME));}
+      } else if (jj_2_15(2)) {
+        e = FunctionCall();
+                                     {if (true) return e;}
+      } else {
+        switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+        case S_IDENTIFIER:
+        case S_QUOTED_IDENTIFIER:
+          s = TableColumn();
+                       {if (true) return new MConstant(s, MConstant.COLUMNNAME);}
+          break;
+        case S_NUMBER:
+          t = jj_consume_token(S_NUMBER);
+                    {if (true) return new MConstant(t.toString(), MConstant.NUMBER);}
+          break;
+        case S_CHAR_LITERAL:
+          t = jj_consume_token(S_CHAR_LITERAL);
+     s = t.toString();
+     if(s.startsWith("\u005c'")) s = s.substring(1);
+     if(s.endsWith("\u005c'")) s = s.substring(0, s.length()-1);
+     {if (true) return new MConstant(s, MConstant.STRING);}
+          break;
+        case S_BIND:
+          t = jj_consume_token(S_BIND);
+                  {if (true) return new MConstant(t.toString(), MConstant.STRING);}
+          break;
+        case 88:
+          jj_consume_token(88);
+          e = SQLExpression();
+          jj_consume_token(90);
+                                 {if (true) return e;}
+          break;
+        default:
+          jj_la1[95] = jj_gen;
+          jj_consume_token(-1);
+          throw new ParseException();
+        }
+      }
+    }
+    throw new Error("Missing return statement in function");
+  }
+
+// AggregateFunc ::= SUM | AVG | MAX | MIN | COUNT
+// 01/07/00 PY.Gibello - Added COUNT
+  static final public String AggregateFunc() throws ParseException {
+  Token t;
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case K_SUM:
+      t = jj_consume_token(K_SUM);
+              {if (true) return t.toString();}
+      break;
+    case K_AVG:
+      t = jj_consume_token(K_AVG);
+              {if (true) return t.toString();}
+      break;
+    case K_MAX:
+      t = jj_consume_token(K_MAX);
+              {if (true) return t.toString();}
+      break;
+    case K_MIN:
+      t = jj_consume_token(K_MIN);
+              {if (true) return t.toString();}
+      break;
+    case K_COUNT:
+      t = jj_consume_token(K_COUNT);
+                {if (true) return t.toString();}
+      break;
+    default:
+      jj_la1[96] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+    throw new Error("Missing return statement in function");
+  }
+
+  static final public MExpression FunctionCall() throws ParseException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.